core/src/main/java/org/apache/calcite/sql/SqlWindow.java

```diff
@@ -660,6 +664,12 @@ public boolean isAllowPartial() {
     }
   }
 
+  private boolean onlyLiteralBounds(@Nullable SqlNode lowerBound, @Nullable SqlNode upperBound) {
```

Comment:
This does not seem to check for what the compiler calls "literal", which is essentially a constant value.
You are checking for specific kinds of bounds. So perhaps this function is not named properly.

Reply:
Naming is hard.. My thought was literal in the sense of `SqlLiteral`, but I can see why it's confusing. 
Changed it to `onlySymbolBounds`.